### PR TITLE
OJS3 PorticoPlugin

### PR DESCRIPTION
**Pull request by RahulChakravarthy**
Hi James, this is the portico plugin ported to OJS3. A few notes: functionality-wise, the plugin exports and ftp transfers both single and multiple issues for a particular journal. All the front end links function properly (the styling is kind of off but that is an easy fix). Overall everything seems to work, my only issue (you'll see this as you read the code) is that I had to comment out the HTML Gallery section simply because I could not find the equivalent implementation (i.e classes/methods) in OJS3 and it needed to be commented out to prevent it the plugin from potentially breaking. Everything else should work, if you need me to make any immediate changes just let me know. Thanks.

### CODE REVIEW

File: PorticoExportDom.inc.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ class PorticoExportDom {
 	function &generateArticleDom(&$doc, &$journal, &$issue, &$section, &$article) {
 
 		// register the editor submission DAO for use later
-		$editorSubmissionDao =& DAORegistry::getDAO('EditorSubmissionDAO');
+		//$editorSubmissionDao =& DAORegistry::getDAO('EditorSubmissionDAO');
```
Comment (jmacgreg) [inline PorticoExportDom.inc.php]: If this is unnecessary, maybe just delete entirely?

File: PorticoExportDom.inc.php [hunk 2]
```diff
@@ -140,41 +140,43 @@ function &generateArticleDom(&$doc, &$journal, &$issue, &$section, &$article) {
...
-	
-			if ($isHtml) {
-				$styleFile = $galley->getStyleFile();
-				if ($styleFile) {
-					$styleNode =& XMLCustomWriter::createChildWithText($doc, $articleMetaNode, 'related-object');
-					XMLCustomWriter::setAttribute($styleNode, 'xlink:href', $styleFile->getFileName());
-					XMLCustomWriter::setAttribute($styleNode, 'content-type', $styleFile->getFileType());
-					$objectNode =& XMLCustomWriter::createChildWithText($doc, $styleNode, 'object-id', $styleFile->getFileName());
-				}
+			// check for HTML-ness, and if there are additional galley files include them **FIGURE OUT WHAT THE REPLACEMENT METHODS FOR THIS ARE**
```
Comment (jmacgreg) [inline PorticoExportDom.inc.php]: There's a note here that probably needs to be deleted. If it's still a question that needs answering, you can ask either myself, Michael or Alec.

Comment (jmacgreg) [reply PorticoExportDom.inc.php]: Ah - I see now, this looks like the section you were referring to that had to be un-commented-out. If you need a hand with any of this, just let me know.

File: first.diff [hunk 3]
```diff
@@ -0,0 +1,42 @@
+diff --git a/PorticoExportDom.inc.php b/PorticoExportDom.inc.php
```
Comment (jmacgreg) [inline first.diff]: This file should be removed from the commit.

File: p.diff [hunk 4]
```diff
@@ -0,0 +1,120 @@
+diff --git a/PorticoExportDom.inc.php b/PorticoExportDom.inc.php
```
Comment (jmacgreg) [inline p.diff]: This file should be removed from the commit.

File: settingsForm.tpl [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{**
+ {**
```
Comment (jmacgreg) [inline settingsForm.tpl]: The added space should be removed.

**Comment by RahulChakravarthy** [comment]
Hi @jmacgreg, I have completed the changes you asked for and tested everything. I ended up removing the html galley and image files code since I'm fairly certain they are being added to the archives during the add Supplementary files portion of the method. Furthermore when I export the issues either by downloading or ftp, I can see html files in the zip file meaning they are being added. Let me know if there are any further changes or testing that you would like done. Thanks

**Comment by jmacgreg** [comment]
Thanks Rahul! I'm not seeing the new commit though - do you need to close and send a new request, or do I need to do something?

**Pull request closed without merging**